### PR TITLE
RLT: Adds In Code Utils

### PR DESCRIPTION
**Pull request by klaurtar**
Adds in the initial code utilites to help Amplify Cognito migrations.

### CODE REVIEW

File: cognito-csv-exporter.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const params = {
+    UserPoolId: userPoolId,
+  };
+
+  const allUsers = await getAllUsers(params);
+
+  // provide the user pool ID of the new pool you want to transfer users to
+  cognitoIdentityServiceProvider.getCSVHeader(
+    {
+      UserPoolId: 'example-user-pool-id',
```
Comment (MixMasterT) [inline cognito-csv-exporter.js]: For convenience, I would suggest moving all of these strings that are meant to be replaced by the user into consts up at the top of the file. I would use ALL_CAPS for them, and make it obvious that their values are meant to be set manually before running the script

File: copyDynamoTable.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const { copy } = require('copy-dynamodb-table');
+
+const globalAWSConfig = {
+  // AWS Configuration object http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#constructor-property
+  accessKeyId: 'your_access_key',
```
Comment (MixMasterT) [inline copyDynamoTable.js]: same here with these variables.

Also, I see this `gobalAWSConfig` used in a couple of places. Would it be worth moving that to its own file?

File: copyDynamoTable.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      tableName: destinationTable, // required
+    },
+    log: true, // default false
+    create: false, // create destination table if not exist
+  });
+}
+
+async function init() {
+  try {
+    await copyTable('example-source-table', 'example-destination-table');
```
Comment (MixMasterT) [inline copyDynamoTable.js]: these string constants could also be treated as described in my other comment.

File: csv-utils/userToCsv.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+exports.userToCsvHeader = (users) => {
+  const convertedUsers = [];
+
+  for (let i = 0; i < users.length; i += 1) {
+    const user = users[i];
+    const { Username, Attributes, UserLastModifiedDate } = user;
+    console.log(user);
```
Comment (MixMasterT) [inline csv-utils/userToCsv.js]: Why log this user here? If it is intentional, I would suggest filling the logged statement in with a bit more information.

File: csv-utils/userToCsv.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    const dateTime = new Date(UserLastModifiedDate);
+
+    // Update this to transfer whatever attributes are needed from the previous user pool to the new pool
+
+    if (Attributes[1].Value === 'true') {
+      convertedUsers.push([
+        '',
```
Comment (MixMasterT) [inline csv-utils/userToCsv.js]: I believe a comment here about the rigid structure of the expected CVS would be helpful.
Also, I would love to see a line-by-line comment for what each of these strings represents (that might be pretty tedious though...)

File: deleteCognitoUsers.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# A simple script to delete all users from a cognito user pool. NOTE: jq must be installed
+
+COGNITO_USER_POOL_ID=your-user-pool-id
```
Comment (MixMasterT) [inline deleteCognitoUsers.sh]: This is just like what I was suggesting with the configurable consts up near the top of the file.

File: updateTablesToNewUser.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+  accessKeyId: 'your_access_key',
+  secretAccessKey: 'your_secret_access_key',
+  region: 'your_region',
+});
+
+// Create an instance of the AWS Cognito Identity provider client
+const cognitoIdentityServiceProvider = new AWS.CognitoIdentityServiceProvider();
+
+// The ID of the Cognito User Pool you want to retrieve users from
+const userPoolId = 'us-east-1_lfq00ab2u';
```
Comment (MixMasterT) [inline updateTablesToNewUser.js]: Is this an actual userPoolId ? If so, I suggest removing it.

File: updateTablesToNewUser.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+      },
+    },
+    ReturnValues: 'UPDATED_NEW',
+  };
+
+  try {
+    // Call the DynamoDB updateItem method to update the cognitoUsername field
+    await dynamoDB.updateItem(params).promise();
+  } catch (err) {
+    console.log('This is a unsuccessful upload', err);
```
Comment (MixMasterT) [inline updateTablesToNewUser.js]: This error message is not clear to me. What was being uploaded? It seems to me more like
```
update to dynamoDB item ${param.something} failed: err
```

File: cognito-csv-exporter.js (hunk 1 above)
Comment (klaurtar) [reply cognito-csv-exporter.js]: Good idea. I implemented this.

File: copyDynamoTable.js (hunk 2 above)
Comment (klaurtar) [reply copyDynamoTable.js]: Moved to another file

File: copyDynamoTable.js (hunk 3 above)
Comment (klaurtar) [reply copyDynamoTable.js]: UPDATED

File: csv-utils/userToCsv.js (hunk 4 above)
Comment (klaurtar) [reply csv-utils/userToCsv.js]: Removed console.log

File: csv-utils/userToCsv.js (hunk 5 above)
Comment (klaurtar) [reply csv-utils/userToCsv.js]: Added in a comment

File: deleteCognitoUsers.sh (hunk 6 above)
Comment (klaurtar) [reply deleteCognitoUsers.sh]: Implemented

File: updateTablesToNewUser.js (hunk 7 above)
Comment (klaurtar) [reply updateTablesToNewUser.js]: Oops, I removed this out of the commit entirely.

File: updateTablesToNewUser.js (hunk 8 above)
Comment (klaurtar) [reply updateTablesToNewUser.js]: Made a better error message

**Pull request merged**